contrib/hashicorp/consul/consul.go

```diff
@@ -24,12 +25,17 @@ func init() {
 	telemetry.LoadIntegration(componentName)
 }
 
+// A kafkaConfig struct holds information from the kafka config for span tags
+type consulConfig struct {
+	hostname string
+}
+
 // Client wraps the regular *consul.Client and augments it with tracing. Use NewClient to initialize it.
 type Client struct {
 	*consul.Client
-
-	config *clientConfig
-	ctx    context.Context
+	consulConfig *consulConfig
```

Comment:
Let's not have a whole separate config object for the hostname.

Reply:
So these PRs are actually part of a chain and if we see PR #1979, we do something similar where we create a separate config struct to hold **at the moment** hostname, but in the future there are key pieces of information found only in the config upon initialization that we may want to save for improved tagging.

Reply:
Done